app/controllers/users_controller.rb

```diff
@@ -1,10 +1,37 @@
 # frozen_string_literal: true
 
 class UsersController < ApplicationController
+  before_action :verify_current_user, only: %i[edit update]
+
   def show
     @user = User.find_by(nickname: params[:nickname])
     return redirect_to root_path, alert: 'User not found' unless @user
 
     @posts = @user.posts.ordered_by_newest
   end
+
+  def edit
+    @user = User.find_by(id: params[:id])
```

Comment:
This line is in the edit and the update, you can move it to a `before_action`
- By Enz